### DISCUSSION TITLE
Issue with window placement with dual screen.

### DISCUSSION THREAD

**Discussion by chicco-carone** [ppy/osu#25242] [Q&A]
### Type

Crash to desktop

### Bug description

I just updated to the latest version of the game from the aur (osu-lazer-bin) and when i first started the game it opened on my main monitor and after i pressed the windows key to bring the window down when i reopened it it opened on the second display. So i tried moving it with my window manager settings (kwin) and i managed do put it again on the primary monitor but the issue happened again. so i checked the in game settings and the game said that it was on running on the main display even if was on the second screen and if i tried switching displays via the in game settings it started changing my display settings to a point where i needed to unplug it and plug it back in to reload the settings and make it work again.

System info via lutris
[System]
OS:              Arch Linux rolling n/a
Arch:            x86_64
Kernel:          6.5.8-zen1-1-zen
Desktop:         KDE
Display Server:  x11

[CPU]
Vendor:          GenuineIntel
Model:           11th Gen Intel(R) Core(TM) i5-11600K @ 3.90GHz
Physical cores:  6
Logical cores:   12

[Memory]
RAM:             15.4 GB
Swap:            4.0 GB

[Graphics]
Vendor:          AMD
OpenGL Renderer: AMD Radeon RX 5700 XT (navi10, LLVM 16.0.6, DRM 3.54, 6.5.8-zen1-1-zen)
OpenGL Version:  4.6 (Compatibility Profile) Mesa 23.2.1-arch1.2
OpenGL Core:     4.6 (Core Profile) Mesa 23.2.1-arch1.2
OpenGL ES:       OpenGL ES 3.2 Mesa 23.2.1-arch1.2
Vulkan Version:  1.3.269
Vulkan Drivers:  AMD Radeon RX 5700 XT (RADV NAVI10) (1.3.255)

### Screenshots or videos

what happens after i change screens
![IMG20231026153934](https://github.com/ppy/osu/assets/66080458/3b19ff7e-8f04-4e7e-8643-9c4a1bbeb48c)

the game that doesn't want to change screen
https://github.com/ppy/osu/assets/66080458/6000f83c-1b04-4835-8ddb-4a6d4a6f3c40

### Version

2023.1026.0

### Logs

Unfortunately i can't get the log file because it completely blocks every input until i close the game and i do it via ssh. I'll try to get the logs via ssh and upload them

**Comment by Sclafus** [comment]
The issue is not really explained well, but I can reproduce it with a similar environment. (3500x + GTX 1070 Driver Version: 535.113.01 on Arch Linux, latest kernel)

Steps:
- set osu! to launch on the primary display
- in the osu! settings, switch the display the secondary one
- still in the osu! settings, switch the display to the primary one
- primary display will get disabled in the settings

**Comment by chicco-carone** [comment]
I can add that after trying on wayland the issue can't be reproduced with the same steps
